src/components/TopicSearch.jsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { getTopics } from "./utils/utils";
+
+const TopicSearch = ({ onChange }) => {
+    const [topics, setTopics] = useState([]);
+    const [selectedTopic, setSelectedTopic] = useState("");
+    const [isLoading, setIsLoading] = useState(true);
+    const [isError, setIsError] = useState(false);
+  
+    useEffect(() => {
+      setIsLoading(true);
+      setIsError(false);
+  
+      getTopics()
+        .then((topics) => {
+          setTopics(topics);
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          console.error("Error fetching topics:", error);
+          setIsError(true);
+          setIsLoading(false);
+        });
+    }, []);
+  
+    const handleChange = (e) => {
```

Comment:
This task says the following: 

- Ensure that you are are updating the URL path for each topic.
- A user should be able share the URL with a friend and only fetch the articles associated with that topic.

To achieve this, you will need to use the Link tag and paths so your URL will reflect the selected topic.